src/pip/_internal/utils/misc.py

```diff
@@ -765,15 +765,33 @@ def unpack_file(
 
 
 def format_command_args(args):
-    # type: (List[str]) -> str
+    # type: (Union[List[str], List[Union[str, HiddenText]]]) -> str
     """
     Format command arguments for display.
     """
-    return ' '.join(shlex_quote(arg) for arg in args)
+    # Arguments can be HiddenText, in which case the raw value should be
+    # redacted when being displayed. This is done through a cast to str.
+    # py27: Only cast to str if the argument type is an instance of HiddenText.
+    # Casting to a str when a non-ascii character is present and the type(arg)
+    # is unicode can trigger a UnicodeDecodeError.
+    return ' '.join(
+        shlex_quote(str(arg)) if isinstance(
```

Comment:
I think the prior comment is still pending, can be `shlex_quote(str(arg)) for arg in args`.

Reply:
@chrahunt Thanks for the feedback, I actually originally edited this to be `shlex_quote(str(arg)) for arg in args`.

@cjerdonek [requested](https://github.com/pypa/pip/pull/6862#issuecomment-520327996) here that I change to use `isinstance`.

I tried to create a `UnicodeDecodeError` appear in this function using py27 but was unable to. I think it's ok to program defensively. I also think this is unlikely to fail in actual operation. 😄 

Based on the feedback though, I'm not sure if I should leave the `isinstance` check in or take it out

Reply:
I'd rather not risk introducing errors that we don't need to risk.

Reply:
Thanks, sorry I had missed that comment.